### PR TITLE
Move Transaction_logic out of Mina_base

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR moves `Transaction_logic` to a dedicated `Mina_transaction_logic` module.

This is some housekeeping prework to allow #10520 to be implemented without injecting a large number of new modules into `Mina_base`.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

File: src/lib/mina_ledger/sparse_ledger.ml [hunk 1]
```diff
@@ -77,3 +78,36 @@ let%test_unit "of_ledger_subset_exn with keys that don't exist works" =
       [%test_eq: Ledger_hash.t]
         (Ledger.merkle_root ledger)
         ((merkle_root sl :> Random_oracle.Digest.t) |> Ledger_hash.of_hash))
+
+module T = Mina_transaction_logic.Make (L)
```
Comment (mitschabaude) [inline src/lib/mina_ledger/sparse_ledger.ml:82]: I think this transaction logic is needed by snarkyjs for its "Mock blockchain" interface (which is important to maintain as a way for developers to test their zk apps locally). Moving this into mina_ledger makes it inaccessible to snarkyjs.

So, either this has to be moved somewhere it remains accessible for JS compilation, or it's simple enough to just be duplicated for snarkyjs!

Comment (mitschabaude) [reply src/lib/mina_ledger/sparse_ledger.ml:82]: (Leaving another comment so I can "request changes")

**Review by mitschabaude — CHANGES_REQUESTED** [review]

File: src/lib/mina_ledger/sparse_ledger.ml (hunk 1 above)
Comment (mrmr1993) [reply src/lib/mina_ledger/sparse_ledger.ml:82]: > So, either this has to be moved somewhere it remains accessible for JS compilation, or it's simple enough to just be duplicated for snarkyjs!

You should still be able to use it just as before; the only difference is you'll need to add `mina_transaction` in your dune file and `open Mina_transaction` in the `.ml` file

Comment (mitschabaude) [reply src/lib/mina_ledger/sparse_ledger.ml:82]: Yeah sorry, I was wrong... snarkyjs uses Transaction_logic and not Sparse_ledger_base

**Review by mitschabaude — APPROVED** [review]

File: src/lib/mina_ledger/sparse_ledger.ml (hunk 1 above)
Comment (mitschabaude) [reply src/lib/mina_ledger/sparse_ledger.ml:82]: This realization would have simplified my earlier refactorings to mina_ledger a lot :D

**Review by bkase — APPROVED** [review]

**Pull request merged**